src/components/Modal/Modal.android.tsx

```diff
@@ -34,14 +42,34 @@ const Modal = ({ visible, children, onRequestClose }: Props) => {
     };
   }, []);
 
-  return <>{visible && <View style={styles.root}>{children}</View>}</>;
+  if (!visible) {
+    return null;
+  }
+
+  const statusBarHidden = visible && presentationStyle === "overFullScreen";
```

Comment:
Here `visible` is always `true` so could be omitted

Reply:
Good point @antonKalinin , refactoring after testing is an unavoidable 😈